src/daemon/bitcoind/actions.rs

```diff
@@ -809,6 +851,11 @@ fn update_utxos(
                     outpoint
                 ))
             })?;
+        // If it spends the unvaulting and goes back to the deposit address, this is a cancel
```

Comment:
What do you think of rather phrasing it like this: "if the spending transaction `txid` fulfills the predicate `txid == cancel_txid` then it's a Cancel" ? :)

The logic i was thinking about is:
- Is it an Emergency?
- If not, Is it a Cancel?
- If not, fetch the unknown txid and treat it as a Spend.

The operation above (`get_spender_txid`) is also quite expensive in real conditions so we should check if it's a Cancel beforehand.